beacon-chain/p2p/service.go

```diff
@@ -146,6 +152,9 @@ func (s *Service) Start() {
 		return
 	}
 
+	// Wait until the beacon state is initialized.
+	s.awaitStateInitialized()
```

Comment:
Now we wait for the state initialized event to trigger before we can launch the p2p service

Reply:
What about pre-genesis? Will this affect nodes awaiting chain start?

Reply:
Yes, we will not start the p2p service until we can get the genesis validator root from the state

Reply:
+1

Reply:
P2P needs access to the genesis time to start according to spec - we can't peer with others unless we have a genesis time

Reply:
Just to clarify my question: will we be able to peer before the chain starts?

I.e. once we hit the deposit threshold, we have a chain start time of midnight of the next day, but the genesis validator set can still grow. Can we peer?